src/state_machine/signer/mod.rs

```diff
@@ -577,7 +577,22 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
             .map(|nr| nr.signer_id)
             .collect::<Vec<u32>>();
 
-        debug!("Got SignatureShareRequest for signer_ids {:?}", signer_ids);
+        let signer_id_set = sign_request
+            .nonce_responses
+            .iter()
+            .map(|nr| nr.signer_id)
+            .collect::<BTreeSet<u32>>();
+
+        if signer_ids.len() != signer_id_set.len()
+            || signer_id_set.is_empty()
+            || signer_id_set.len() > self.total_signers.try_into().unwrap()
+            || signer_id_set.last().unwrap() > &self.total_signers
```

Comment:
Is this check necessary? As in, would having non-sequential `signer_id`s break WSTS?

Reply:
The `FROST` paper specifies sequential party IDs, so we use sequential key IDs and signer IDs, since party IDs are key IDs for v1, and signer IDs for v2.

Reply:
In terms of the Lagrange interpolation, it doesn't care what the IDs are. Having them sequential does make a lot of code more concise.  The Frost coordinator especially takes advantage of this since it doesn't support dkg_threshold.